### PR TITLE
[dagster-dbt] Spec-ify dbt cloud integration

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Updates the dbt cloud integration to be more spec-native, taking advantage of existing utilities for generating specs from a dbt project.

This results in the ability to delete a large amount of code.

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26662?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26662** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26662?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26662?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26614** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26614?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26613** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26613?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#26625](https://github.com/dagster-io/dagster/pull/26625) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26625?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#23824** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23824?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py [hunk 1]
```diff
@@ -890,137 +881,6 @@ def _validate_asset_keys(
         )
 
 
-def get_asset_deps(
```
Comment (dpeng817) [inline python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py:893]: so wonderful

File: python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py [hunk 2]
```diff
@@ -354,67 +343,58 @@ def get_freshness_policy(cls, dbt_resource_props):
...
-            automation_conditions_by_key,
-            _,
-            fqns_by_output_name,
-            metadata_by_output_name,
-        ) = get_asset_deps(
-            dbt_nodes=dbt_nodes,
-            deps=dbt_dependencies,
-            # TODO: In the future, allow the IO manager to be specified.
-            io_manager_key=None,
+        # generate specs for each executed node
```
Comment (dpeng817) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py:346]: filler comment

File: python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py [hunk 3]
```diff
@@ -354,67 +343,58 @@ def get_freshness_policy(cls, dbt_resource_props):
...
+            manifest=manifest_json,
             dagster_dbt_translator=CustomDagsterDbtTranslator(),
-            manifest=None,
+            select=" ".join(
+                f"fqn:{'.'.join(dbt_nodes[unique_id]['fqn'])}" for unique_id in executed_unique_ids
+            ),
         )
 
         return AssetsDefinitionCacheableData(
             # TODO: In the future, we should allow additional upstream assets to be specified.
```
Comment (dpeng817) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py:357]: while we're here can we get rid of this / make a linear ticket (but I think this is possible via map asset specs now, so not really necessary

File: python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py [hunk 4]
```diff
@@ -354,67 +343,58 @@ def get_freshness_policy(cls, dbt_resource_props):
...
-            keys_by_input_name={
-                input_name: asset_key for asset_key, (input_name, _) in asset_ins.items()
-            },
-            keys_by_output_name={
-                output_name: asset_key for asset_key, (output_name, _) in asset_outs.items()
-            },
+            keys_by_output_name={spec.key.to_python_identifier(): spec.key for spec in specs},
```
Comment (dpeng817) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py:358]: somewhat surprised we don't also have to include checks here, but it doesn't look like we did before either

**Review by dpeng817 — APPROVED** [review]
a few comments but generally awesome

**Comment by OwenKephart** [comment]
### Merge activity

* **Dec 30, 9:41 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/26662).

**Pull request merged**